Tests/merge_test.py

```diff
@@ -1,12 +1,247 @@
+		self.table2.table.LookupList.Lookup[0].LookupType = 1 # single lookup type
+		sub = SingleSubst()
+		sub.SubstFormat = 1
+		sub.DeltaGlyphID = 1024
+		self.table2.table.LookupList.Lookup[0].SubTable[0] = sub
+
+		self.mergedTable = ttLib.newTable('GSUB')
+
+	def test_GSUB_merge_no_dups(self):
+		self.merger.duplicateGlyphsPerFont = [{}, {}]
+
+		self.mergedTable.merge(self.merger, [self.table1, self.table2])
+		table = self.mergedTable.table
+
+		# Checks ScriptList
+		self.assertEqual(table.ScriptList.ScriptCount, 2)
+		self.assertIn(self.table1.table.ScriptList.ScriptRecord[0], table.ScriptList.ScriptRecord)
+		self.assertIn(self.table2.table.ScriptList.ScriptRecord[0], table.ScriptList.ScriptRecord)
+
+		# Checks LookupList
+		self.assertEqual(table.LookupList.LookupCount, 2)
+		self.assertIn(self.table1.table.LookupList.Lookup[0], table.LookupList.Lookup)
+		self.assertIn(self.table2.table.LookupList.Lookup[0], table.LookupList.Lookup)
+
+		# Checks FeatureList
+		self.assertEqual(table.FeatureList.FeatureCount, 2)
+		self.assertIn(self.table1.table.FeatureList.FeatureRecord[0], table.FeatureList.FeatureRecord)
+		self.assertIn(self.table2.table.FeatureList.FeatureRecord[0], table.FeatureList.FeatureRecord)
+
+	def test_GSUB_merge_with_dups(self):
+		self.merger.duplicateGlyphsPerFont = [{}, {'uni0032#0': 'uni0032#1'}]
+		self.preMerge(self.table1)
+		self.preMerge(self.table2)
+
+		self.mergedTable.merge(self.merger, [self.table1, self.table2])
+		table = self.mergedTable.table
+
+		# Checks ScriptList
+		self.assertEqual(table.ScriptList.ScriptCount, 2)
+		self.assertIn(self.table1.table.ScriptList.ScriptRecord[0], table.ScriptList.ScriptRecord)
+		self.assertIn(self.table2.table.ScriptList.ScriptRecord[0], table.ScriptList.ScriptRecord)
+
+		# Checks LookupList
+		self.assertEqual(table.LookupList.LookupCount, 3)
+		self.assertIn(self.table1.table.LookupList.Lookup[0], table.LookupList.Lookup)
+		self.assertIn(self.table2.table.LookupList.Lookup[0], table.LookupList.Lookup)
+		self.assertEqual(table.LookupList.Lookup[-1].LookupFlag, 0)
+		self.assertEqual(table.LookupList.Lookup[-1].LookupType, 1)
+		self.assertEqual(table.LookupList.Lookup[-1].SubTableCount, 1)
+		self.assertEqual(table.LookupList.Lookup[-1].SubTable[0].mapping, self.merger.duplicateGlyphsPerFont[1])
+
+		# Checks FeatureList
+		self.assertEqual(table.FeatureList.FeatureCount, 3)
+		self.assertIn(self.table1.table.FeatureList.FeatureRecord[0], table.FeatureList.FeatureRecord)
+		self.assertIn(self.table2.table.FeatureList.FeatureRecord[0], table.FeatureList.FeatureRecord)
+		self.assertEqual(table.FeatureList.FeatureRecord[-1].FeatureTag, 'locl')
+
+	def test_GSUB_merge_with_same_dups_and_no_second_GSUB(self):
+		self.table2 = NotImplemented
+		self.merger.duplicateGlyphsPerFont = [{}, {'uni0032#0': 'uni0032#1'}]
+		self.merger.fonts = []
+		fontTools.merge.isGlyphSame = lambda fonts, ogid, gid : True
```

Comment:
hm, this will persistently set the `merge` module's `isGlyphSame` attribute to the lambda function..
Use a `try: ... finally: ...` or a `@contextlib.contextmanager` to restore the original value at the end.